sklearn/kernel_approximation.py

```diff
@@ -676,25 +679,9 @@ def fit(self, X, y=None):
             Returns the transformer.
         """
         self._validate_params()
-
         X = self._validate_data(X, accept_sparse="csr")
         check_non_negative(X, "X in AdditiveChi2Sampler.fit")
 
-        if self.sample_interval is None:
-            # See reference, figure 2 c)
-            if self.sample_steps == 1:
-                self.sample_interval_ = 0.8
-            elif self.sample_steps == 2:
-                self.sample_interval_ = 0.5
-            elif self.sample_steps == 3:
-                self.sample_interval_ = 0.4
-            else:
-                raise ValueError(
-                    "If sample_steps is not in [1, 2, 3],"
-                    " you need to provide sample_interval"
-                )
-        else:
-            self.sample_interval_ = self.sample_interval
```

Comment:
It will be a bit more tricky than that. We need to have a deprecation cycle where we will still compute those during two release cycles.

What I propose is to have the following pattern:

```python
class AdditiveChi2Sampler:
    def fit(self, X, y=None):
        # store the value in private attributes
        self._sample_interval = ...

    @property
    def sample_interval_(self):
        # announce the deprecation with a warning

    def transform(self, X):
        # use self._sample_interval in the body
```

Reply:
We also need to add the deprecation in the docstring.

Reply:
and add some test for checking the warning is raised or that an attribute error is raised when not `fitted`